packages/internals/src/flow.js

```diff
@@ -21,7 +19,8 @@ class Flow {
     return new Flow(this.scope, data);
   }
   map(fn, ...args) {
-    const data = this.data.map(item => {
+    const map = fn.isLazy ? this.data.lazyMap : this.data.map;
+    const data = map.call(this.data, item => {
       let fun = fn instanceof Method ? fn.get(this.data) : fn;
       if (typeof fun == "function") fun = new JSFn(fun);
```

Comment:
This should probably be `===`. If you want, we can turn #134 into a maintainance issue that takes care of multiple things, including this.

Reply:
Yeah, let's do that. We should probably have some helper functions like isObject, isFunction